store/store_action.go

```diff
@@ -566,6 +571,17 @@ func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
 	}
 
 	if resp.StatusCode != 200 {
+		// some fields might not be supported on proxies with old versions.
+		// we should retry with the snap store version known as we can now
+		// get it from the response header.
+		if resp.StatusCode == 400 && storeVer == nil {
+			verstr := resp.Header.Get("Snap-Store-Version")
+			ver, err := strconv.Atoi(verstr)
+			if err != nil {
+				logger.Debugf("Bogus Snap-Store-Version header %q.", verstr)
```

Comment:
When there's an error we usually start our error messages with "cannot". It's not super important here because it's not a user facing error but good to be aware of that pattern. I'd rephrase this because the value is more unexpected or wrong that bogus, something like: `cannot parse header value of Snap-Store-Version: expected int got %q`
Just a suggestion, doesn't have to be specifically that

Reply:
Updated, Thank you!